<a href="https://colab.research.google.com/github/CillaCole/Project-2/blob/main/Project_2%2C_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

# Dataset Choice #1: Mushroom Classification

In [2]:
# source of data
# the data came from: https://www.kaggle.com/datasets/uciml/mushroom-classification
df = pd.read_csv('/content/mushrooms.csv')
df.head()

class cap-shape cap-surface cap-color bruises odor gill-attachment  \
0     p         x           s         n       t    p               f   
1     e         x           s         y       t    a               f   
2     e         b           s         w       t    l               f   
3     p         x           y         w       t    p               f   
4     e         x           s         g       f    n               f   

  gill-spacing gill-size gill-color  ... stalk-surface-below-ring  \
0            c         n          k  ...                        s   
1            c         b          k  ...                        s   
2            c         b          n  ...                        s   
3            c         n          n  ...                        s   
4            w         b          k  ...                        s   

  stalk-color-above-ring stalk-color-below-ring veil-type veil-color  \
0                      w                      w         p          w   
1                      w                      w         p          w   
2                      w                      w         p          w   
3                      w                      w         p          w   
4                      w                      w         p          w   

  ring-number ring-type spore-print-color population habitat  
0           o         p                 k          s       u  
1           o         p                 n          n       g  
2           o         p                 n          n       m  
3           o         p                 k          s       u  
4           o         e                 n          a       g  

[5 rows x 23 columns]

In [9]:
df.shape

(8124, 23)

- Are mushrooms safe or deadly? This data was collected from the Audubon Society Field Guide to North American Mushrooms. There is no simple rule or trick for seeing if mushrooms are poisonous or not, however there are other features that can help us decide. This data includes different species and features of said species. With those features, we should we able to deduce whether that speciies is safe for consumption, or poisonous.
- The target is the class of mushroom.
- This is a classification problem.
- There are 23 features
- There are 8124 rows of data
- Trying to figure out if a species is poisonous or not (without having a one consistent rule) presents its own challenges. I forsee possible challenges in trying to display consistency and the solutions well as a data scientist as well as be able to explain solutions to an audience without a data scientist. 

# Dataset #2: Spanish Wine Dataset

In [4]:
# source the data and display the head
# this data was sourced from: https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset/code?resource=download
dframe = pd.read_csv('/content/wines_SPA.csv')
dframe.head()

winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

In [11]:
dframe.shape

(7500, 11)

- This Spanish dataset displays different characteristics relating to a variety of Spanish red wines. There are many different attributes that play into determining how a bottle of wine should be priced. This dataset represents a conglomerate of these different attributes that were taken from wine specialized pages.
- The target is the price.
- This is a regression problem.
- This dataset has 11 features.
- This dataset has 7500 rows.
- I forsee challenges in choosing, cleaning and modeling the data in a way that communicates the data the way I need it to.